### PR TITLE
Check not frozen before saving DAG

### PR DESCRIPTION
**Pull request by garyluu**
For #4284 

Very unique case where there's a workflow that we didn't save the DAG for and then it was snapshotted before we had the DAG-saving code.

Can't really make a test case easily because the endpoint that sets a frozen version also saves a DAG...i need a version that's frozen but has no DAG saved

### CODE REVIEW

**Review by coverbeck — APPROVED** [review]

**Comment by denis-yuen** [comment]
> Can't really make a test case easily because the endpoint that sets a frozen version also saves a DAG...i need a version that's frozen but has no DAG saved

Can you find a regular test case with a dag, then hit the DB directly like https://github.com/dockstore/dockstore/blob/711b0d78c27653ab02e8457963c3aa49aa960ec7/dockstore-integration-testing/src/test/java/io/dockstore/client/cli/MigrationIT.java#L114 to wipe out the DAG first then set the frozen column to true?

File: dockstore-common/generated/src/main/resources/pom.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>io.dockstore</groupId>
   <artifactId>dockstore-common</artifactId>
-  <version>1.11.0-beta.3</version>
+  <version>1.11.0-beta.4-SNAPSHOT</version>
```
Comment (denis-yuen) [inline dockstore-common/generated/src/main/resources/pom.xml:22]: Ah crud, my bad. I think all these Snyk ones were not checked in

**Review by denis-yuen — APPROVED** [review]

**Comment by garyluu** [comment]
> > Can't really make a test case easily because the endpoint that sets a frozen version also saves a DAG...i need a version that's frozen but has no DAG saved
> 
> Can you find a regular test case with a dag, then hit the DB directly like
> 
> https://github.com/dockstore/dockstore/blob/711b0d78c27653ab02e8457963c3aa49aa960ec7/dockstore-integration-testing/src/test/java/io/dockstore/client/cli/MigrationIT.java#L114
> 
> to wipe out the DAG first then set the frozen column to true?

Is that a really good test though? Seems like a hacky way of freezing a version

**Comment by denis-yuen** [comment]
> Is that a really good test though? Seems like a hacky way of freezing a version

I'd say it wouldn't be a good test for how to freeze a version. 

It's strictly a test to ensure that we avoid a 500 in this database situation now and in the future if someone else tinkers with this and accidentally introduces a regression.

**Review by NatalieEO — APPROVED** [review]

**Comment by denis-yuen** [comment]
Probably bitbucket rate limit issues, going to merge and use as part of deploy

**Pull request merged**